src/main/scala/cromwell/engine/ExecutionTable.scala

```diff
@@ -0,0 +1,11 @@
+package cromwell.engine
+
+import cromwell.binding.{Call, WdlBinding}
+
+/**
+ * Corresponds to the "execution table" of our discussions.
+ */
```

Comment:
Currently these structures do feel rather tabular, though when we discuss references that starts to feel a bit graphy.  I guess we'll see how the choice of implementation plays out, but I thought having 'Table' in the class names too strongly implied a tabular implementation, even if the term was actually meant more colloquially.

For now I've generically named these `SymbolStore` and `ExecutionStatusStore` and made a separate case class `SymbolStoreKey` for the unique tuple of (scope, name, iteration, input/output).  If this seems okay the branch should be good to merge, I already did the medieval cut-new-branch/patch/set-upstream thing with no obvious ill effects.